### PR TITLE
Cross test explicit nulls

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This adds a cross test for explicit null collections on the TF side.
Not sure if we are going to be able to distinguish between null vs unspecified on our side but I thought it was an interesting corner case which might be worth revisiting.

### CODE REVIEW

File: pkg/tests/cross-tests/pretty.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ func (s prettyValueWrapper) GoString() string {
 		tL := tp.TypeReferenceString(v.Type())
 		indent := strings.Repeat("  ", level)
 		switch {
+		case v.IsNull():
+			// TODO
```
Comment (t0yv0) [inline pkg/tests/cross-tests/pretty.go:76]: What does this signify? Thanks!

File: pkg/tests/cross-tests/input_cross_test.go [hunk 2]
```diff
@@ -217,3 +217,35 @@ func TestInputsEmptyConfigModeAttrSet(t *testing.T) {
...
+						Elem: &schema.Schema{
+							Type:      schema.TypeInt,
+							Optional:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
+		},
+		// TODO: How does one express this with tftypes?
```
Comment (t0yv0) [inline pkg/tests/cross-tests/input_cross_test.go:248]: `tftypes.NewValue(typ, nil)` expresses a null value. Every type is inhabited by a null value.

File: pkg/tests/cross-tests/input_cross_test.go [hunk 3]
```diff
@@ -217,3 +217,35 @@ func TestInputsEmptyConfigModeAttrSet(t *testing.T) {
...
 }
+
+func TestExplicitNilList(t *testing.T) {
+	skipUnlessLinux(t)
+	// This is an explicit null on the tf side:
+	// resource "crossprovider_testres" "example" {
+	//     f0 = null
+	// }
+	// This is different to an unspecified value.
+	// We might not be able to distinguish between null and missing collections.
```
Comment (t0yv0) [inline pkg/tests/cross-tests/input_cross_test.go:228]: Have you dug into this to make this a bit more precise?

I think there are cases where the distinction ends up being quite meaningful, like TF diffs can be sensitive to this distinction. If/when we lose the distinction in translation is a small window for issues to creep in.

Comment (t0yv0) [reply pkg/tests/cross-tests/input_cross_test.go:228]: The one interesting case which we definitely need to test and annotate explicitly is handling blocks. If I understand this right blocks cannot receive nulls in TF (or unknowns) and that's structurally prevented, however in Pulumi the projection of blocks will always be able to be called with nulls and unknowns and we have to compensate for that possibility.

**Review by t0yv0 — APPROVED** [review]
OK merging this even if there are remaining open questions.

File: pkg/tests/cross-tests/pretty.go (hunk 1 above)
Comment (VenelinMartinov) [reply pkg/tests/cross-tests/pretty.go]: I was under the impression tftypes.NewValue(typ, nil) didn't work but after your comment I tried it again and it works. My bad, thank you!

File: pkg/tests/cross-tests/input_cross_test.go (hunk 2 above)
Comment (VenelinMartinov) [reply pkg/tests/cross-tests/input_cross_test.go]: Thank you for the comment, I tried it again and it is indeed the case - I must have got something wrong when I tried it last time.

File: pkg/tests/cross-tests/input_cross_test.go (hunk 3 above)
Comment (VenelinMartinov) [reply pkg/tests/cross-tests/input_cross_test.go:233]: The comment here is more about the fact that we might not be able to express nil vs missing in our programs or in the bridge.

We need to do a bit more testing but I strongly suspect that we just don't have a concept of an explicitly nil value vs an unspecified value.

Comment (VenelinMartinov) [reply pkg/tests/cross-tests/input_cross_test.go:233]: Also I'm not sure most languages have a distinction there too - maybe just Typescript?

Comment (VenelinMartinov) [reply pkg/tests/cross-tests/input_cross_test.go:233]: I might be wrong here, I'll double test as the HCL docs do say that an explicit null should be equivalent to an unspecified value.

Perhaps the one exception here is ConfigModeAttr hacks, but that's probably fine.
